taskcluster/utils.py

```diff
@@ -276,6 +317,23 @@ def authenticate(description=None):
     The description will be shown on login.taskcluster.net, if not provided
     a default message with script path will be displayed.
   """
+
+  # Let's first offer to read credentials from a configuration file then
+  # the environment
```

Comment:
This won't work... imagine a user with temp creds in file, and perma creds in env vars... :)

How about we just read from env vars, if we get something we return it... otherwise, we try file, and then we try webbrowser thing...